webapp/src/components/user_settings/video_devices_settings_section.tsx

```diff
@@ -0,0 +1,446 @@
+                        />
+                        <span>{editLabel}</span>
+                    </button>
+                </div>
+                <div className='section-min__describe'>
+                    <span>{description}</span>
+                </div>
+            </div>
+        );
+    }
+
+    return (
+        <section className='section-max form-horizontal'>
+            <h4 className='col-sm-12 section-title'>
+                <span>{title}</span>
+            </h4>
+            <div className='sectionContent col-sm-10 col-sm-offset-2'>
+                <div
+                    tabIndex={-1}
+                    className='setting-list'
+                >
+                    <div className='setting-list-item'>
+                        <Fieldset>
+                            <VideoDevicesSelection
+                                deviceType='videoinput'
+                                devices={devices.filter((device) => device.kind === 'videoinput')}
+                                onSelectionChange={async (opt) => {
+                                    previewStream?.getVideoTracks()[0].stop();
+
+                                    const videoOpts = {
+                                        ...DefaultVideoTrackOptions,
+                                    };
+
+                                    if (opt.value !== '') {
+                                        videoOpts.deviceId = {
+                                            exact: opt.value,
+                                        };
+                                    }
+
+                                    const stream = await navigator.mediaDevices.getUserMedia({
+                                        video: videoOpts,
+                                        audio: false,
+                                    });
+
+                                    setPreviewStream(stream);
+                                }}
+                                ref={videoInputsRef}
+                            />
+                            <VideoPreview
+                                stream={previewStream}
+                                mirror={mirrorVideo}
+                                blurBackground={blurBackground}
+                                blurIntensity={blurIntensity}
+                            />
+                            <CheckBoxContainer>
+                                <input
+                                    type='checkbox'
+                                    id='mirror'
+                                    name='mirror'
+                                    checked={mirrorVideo}
+                                    onChange={e => setMirrorVideo(e.target.checked)}
+                                />
+                                <label htmlFor='mirror'>{formatMessage({defaultMessage: 'Mirror video'})}</label>
+                            </CheckBoxContainer>
+                            {!isFirefox() &&
```

Comment:
Unfortunately, GPU acceleration isn't working yet on Firefox (see https://github.com/google-ai-edge/mediapipe/issues/5879). I tested CPU only but it's just not going to work. Anyhow, this is experimental on top of an already experimental feature.